src/x11/screen.c

```diff
@@ -386,32 +389,13 @@ const struct screen_info *get_active_screen(void)
  */
 static Window get_focused_window(void)
 {
-        Window focused = 0;
-        Atom type;
-        int format;
-        unsigned long nitems, bytes_after;
-        unsigned char *prop_return = NULL;
-        Window root = RootWindow(xctx.dpy, DefaultScreen(xctx.dpy));
-        Atom netactivewindow =
-            XInternAtom(xctx.dpy, "_NET_ACTIVE_WINDOW", false);
-
-        XGetWindowProperty(xctx.dpy,
-                           root,
-                           netactivewindow,
-                           0L,
-                           sizeof(Window),
-                           false,
-                           XA_WINDOW,
-                           &type,
-                           &format,
-                           &nitems,
-                           &bytes_after,
-                           &prop_return);
-        if (prop_return) {
-                focused = *(Window *)prop_return;
-                XFree(prop_return);
-        }
+        Window focused;
+        int ignored;
+
+        XGetInputFocus(xctx.dpy, &focused, &ignored);
 
+        if (focused == None || focused == PointerRoot)
+                focused = 0;
```

Comment:
None is a typedef for 0L, so you can leave that out of the if statement. I would also set focused to None for clarity.  Or are there cases where None != 0L?

Reply:
The typedef might not change but it's always best to do what the library devs expect you to in my experience :). It's not worth it for a 5 character reduction.

Reply:
What do you mean? Are you for or against changing it to `focused = None`. I don't think they accidentally typedef'd None to 0.

Reply:
None is typedef'd to 0L in Xorg, but that isn't guaranteed by the X11 spec. There might be a X11 implementation where None = 1L.

Reply:
Allright then this is fine by me.